### DISCUSSION TITLE
AVR128DA schematic

### DISCUSSION THREAD

**Discussion by ITstreet1** [SpenceKonde/DxCore#537] [General]
Hello,

Is there a basic AVR128DA schematic here? What is the basic electronic setup for this MCU?

**Comment by grandaspanna** [comment]
You can have a perfectly functioning solution with just two bypass capacitors - one on Vdd and one on AVdd. Nothing else required really. Runs up to 24MHz without any clock/crystal across the supported voltage range. Route out the UPDI pin and you can program away!
![image](https://github.com/SpenceKonde/DxCore/assets/54263749/c5a12a66-d3d9-4801-95e8-8a5d2319a3f6)

According to the documentation, Vdd and AVdd are internally connected, so you might even get away without the second bypass capacitor, but better safe than sorry.

**Comment by WestfW** [comment]
For which version of the chip?
The AVRxxDA48 has about the same pinout as the ATmega4809, so you can use 4809 board schematics (ie Nano Every) to get a good idea.

Re: @grandaspanna, schematic: that looks like it's for the 32pin QFP/VQFN, but I'm pretty sure you should connect ALL of the GND pins...  And Datasheet says explicitly `Each supply pin pair (power input pin and ground pin) must have separate decoupling capacitors.`  (Section 4, "Hardware guidelines.")

**Reply by grandaspanna** [reply]
You're absolutely right! I just grabbed the model from SnapEDA into my software, but didn't notice it's missing pin 29!

**Comment by ITstreet1** [comment]
Let's say something like this:
![Schematic AVR128DA32](https://github.com/SpenceKonde/DxCore/assets/30090189/e479f41c-fc76-4e0e-b4d9-9e092d77c116)

UPDI header for burning the bootloader, and UART header for programming just like the Uno. Anything to add or remove?

**Reply by grandaspanna** [reply]
Looks ok to me, but I don't bother with a bootloader for my designs and do all the programming via the UPDI interface. I find it so much simpler. On a few boards that can be updated by the end user, I add a CH340 USB chip, but still go via UPDI. There is a use-case for Optiboot where you need to update flash from your code, but that's not something I've needed.

**Reply by ITstreet1** [reply]
I want to build a test bed, not a specific project. From what I can see, this is almost the same as the ATmega4809 setup or megaAVR 0 in general. This board I am building should be exactly like the Uno. That includes the bootloader, UART bridge, headers, etc.
